download/upgradeRecipe/upgradeRecipe8.adoc

```diff
@@ -50,4 +50,176 @@ Any backwards incompatible changes are annotated with a [.label.label-danger.lab
 Due to the limited usage of OSGi and the maintenance burden it brings,
 the OptaPlanner jars in the 8.x series no longer include OSGi metadata in their `META-INF/MANIFEST.MF` file.
 
+[.upgrade-recipe-major]
+=== `filterClassList` replaced by a single filterClass
 
+This change applies to a configuration of `EntitySelector`, `ValueSelector` and `MoveSelector` in both
+the configuration API and the solver configuration XML.
+
+==== Reason:
+In practice, you don't need multiple selection filter classes often, and you can always replace them by a single selection
+filter class which implements the logic of all of them.
+Passing a single selection class now requires less boilerplate code.
+
+Before in `*.java`:
+[source, java]
+----
+ValueSelectorConfig valueSelectorConfig = new ValueSelectorConfig();
+valueSelectorConfig.setFilterClassList(Collections.singletonList(MySelectionFilterClass.class));
+----
+
+After in `*.java`:
+[source, java]
+----
+ValueSelectorConfig valueSelectorConfig = new ValueSelectorConfig();
+valueSelectorConfig.setFilterClass(MySelectionFilterClass.class);
+----
+
+==== Replacing multiple selection filter classes with a single one
+
+Before in `\*.xml` and `*.java`:
```

Comment:
(very soft, for consistency) In other upgrade recipe notes, I tended to just repeat the Before line twice:

Before in `*.xml`:

// xml code

Before in `*.java`:

// java code